### PR TITLE
Pytest/fan

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

This PR focuses on pytests for fan and grid plotting methods 
Note PR #184 should be merged first since this one includes it

**issue:** #179 

## Approval

**Number of approvals:** 2 - code review and testing 
## Test

**matplotlib version**: 3.3.4
**Note testers: please indicate what version of matplotlib you are using**

``` bash
cd pydarn/test/
pytest test_Fan.py 
pytest test_Grid.py
```

### CODE REVIEW

File: test/test_Grid.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+class TestGrid_defaults:
+
+    def test_grid_defaults(self):
+        """ """
+        with warnings.catch_warnings(record=True):
+            pydarn.Grid.plot_grid(data)
+
+@pytest.mark.parametrize('colorbar', [False])
+@pytest.mark.parametrize('colorbar_label', 'green')
```
Comment (carleyjmartin) [inline test/test_Grid.py:34]: I don't know why this line doesn't throw up the error like in test_RTP.py.

**Review by carleyjmartin — COMMENTED** [review]
Cloned and installed branch, ran `pytest test_Fan.py` and `pytest test_Grid.py`
Fan ran without issue, grid has a typo in the data path, once amended no issues with grids.
These tests have 25 and 21 items to test, but the rtp test had >1000, is that okay?

Couple of warnings
```
/lib/python3.7/site-packages/pydarnio/utils/conversions.py: 42: DeprecationWarning: `np.str` is a deprecated alias for the builtin `str`. To silence this warning, use `str` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.str_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

 /lib/python3.7/site-packages/tables/__init__.py: 99: DeprecationWarning: `np.typeDict` is a deprecated alias for `np.sctypeDict`.

  /lib/python3.7/site-packages/pydarn/utils/conversions.py: 44: DeprecationWarning: `np.str` is a deprecated alias for the builtin `str`. To silence this warning, use `str` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.str_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

These same warnings on same lines also found in the RTP plots too.

File: test/test_Grid.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import bz2
+import datetime as dt
+import matplotlib.pyplot as plt
+import pytest
+import warnings
+
+import pydarn
+
+data = pydarn.SuperDARNRead('data/test.grid').read_grid()
```
[the remark this answers is not in the document]
Comment (billetd) [reply test/test_Grid.py:23]: This also fixed the test_Grid.py Pytest for me

**Comment by billetd** [comment]
Have/had no issues running `test_Fan`. Ran into same issue as @carleyjmartin on `test_Grid`.

**Review by carleyjmartin — APPROVED** [review]

**Pull request merged**